lib/logstash/filters/geoip.rb

```diff
@@ -189,4 +181,15 @@ def get_geo_data_for_ip(ip)
       geo_data
     end
   end
+
+  def ensure_database!
+    # Use thread-local access to GeoIP. The Ruby GeoIP module forces a mutex
+    # around access to the database, which can be overcome with :pread.
+    # Unfortunately, :pread requires the io-extra gem, with C extensions that
+    # aren't supported on JRuby. If / when :pread becomes available, we can stop
+    # needing thread-local access.
+    if !Thread.current.key?(@threadkey)
+      Thread.current[@threadkey] = ::GeoIP.new(@database)
```

Comment:
This should be enough no, since `Thread#current` access should be thread safe?

```ruby
 Thread.current[@threadkey] ||= ::GeoIP.new(@database)
```